bin/elasticurl/main.c

```diff
@@ -103,6 +104,7 @@ static void s_usage(int exit_code) {
     fprintf(stderr, "  -t, --trace FILE: dumps logs to FILE instead of stderr.\n");
     fprintf(stderr, "  -v, --verbose: ERROR|INFO|DEBUG|TRACE: log level to configure. Default is none.\n");
     fprintf(stderr, "      --version: print the version of elasticurl.\n");
+    fprintf(stderr, "      --http2: try to use HTTP/2");
```

Comment:
if we're mimicking curl behavior, it just sends h2;http/1.1 every time, if enabled. Why not do that here. And if user wants something else they can speciify http/1.1 explicitly?

Reply:
curl -i -v https://www.google.com
* Rebuilt URL to: https://www.google.com/
*   Trying 172.217.14.228...
* TCP_NODELAY set
* Connected to www.google.com (172.217.14.228) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1


curl --help | grep http
 -0, --http1.0       Use HTTP 1.0 (H)
     --http1.1       Use HTTP 1.1 (H)
     --http2         Use HTTP 2 (H)
     --http2-prior-knowledge  Use HTTP 2 without HTTP/1.1 Upgrade (H)

notice default is to use h2. then you can specify specifically if you want fine grain control.

Reply:
> We can make HTTP/2 happen by default (alpn="h2;http/1.1") once it's feature-complete. For now its opt-in.